pkg/analysis/passes/sourcecode/sourcecode.go

```diff
@@ -8,25 +8,46 @@ import (
 )
 
 var (
-	sourceCodeNotProvided     = &analysis.Rule{Name: "source-code-not-provided", Severity: analysis.Warning}
-	sourceCodeNotFound        = &analysis.Rule{Name: "source-code-not-found", Severity: analysis.Error}
-	sourceCodeVersionMisMatch = &analysis.Rule{Name: "source-code-version-mismatch", Severity: analysis.Error}
+	sourceCodeNotProvided = &analysis.Rule{
+		Name:     "source-code-not-provided",
+		Severity: analysis.Warning,
+	}
+	sourceCodeNotFound = &analysis.Rule{
+		Name:     "source-code-not-found",
+		Severity: analysis.Error,
+	}
+	sourceCodeVersionMisMatch = &analysis.Rule{
+		Name:     "source-code-version-mismatch",
+		Severity: analysis.Error,
+	}
 )
 
 var Analyzer = &analysis.Analyzer{
 	Name:     "sourcecode",
 	Requires: []*analysis.Analyzer{metadata.Analyzer},
 	Run:      run,
-	Rules:    []*analysis.Rule{sourceCodeNotFound, sourceCodeVersionMisMatch, sourceCodeNotProvided},
+	Rules: []*analysis.Rule{
+		sourceCodeNotFound,
+		sourceCodeVersionMisMatch,
+		sourceCodeNotProvided,
+	},
 }
 
 func run(pass *analysis.Pass) (interface{}, error) {
 
-	sourceCodeDir := pass.SourceCodeDir
+	sourceCodeDir := pass.CheckParams.SourceCodeDir
```

Comment:
`pass.CheckParams` could potentially be `nil`. I would probably change `pass.CheckParams` to be a value rather than a pointer

Reply:
I changed it to a value!